backend/routes/userRoutes.js

```diff
@@ -1,93 +1,92 @@
-const bcrypt = require("bcrypt");
-
-router.get("/", async (req, res) => {
-  const users = await prisma.user.findMany();
-  res.json(users);
-});
+const express = require('express')
+const router = express.Router()
+const { PrismaClient } = require('@prisma/client')
+const prisma = new PrismaClient()
+const bcrypt = require('bcrypt');
+
+router.get('/', async (req, res) => {
+  const users = await prisma.user.findMany()
+  res.json(users)
+})
 
-router.post("/", async (req, res) => {
-  const { username, email, password } = req.body;
+router.post('/', async (req, res) => {
+  const {  username, email, password  } = req.body
 
-  const existingUser = await prisma.user.findFirst({
-    where: { username: username },
-  });
+  const existingUser = await prisma.user.findFirst({ where: { username: username } });
   if (existingUser) {
-    return res.status(400).json({ error: "Username already exists" });
+    return res.status(400).json({ error: 'Username already exists' });
   }
 
   const hashedPassword = await bcrypt.hash(password, 10);
   const hashedPasswordValue = String(hashedPassword);
-
+  
   const newUser = await prisma.user.create({
     data: {
-      username,
-      email,
-      password: hashedPasswordValue,
-    },
-  });
+        username, 
+        email, 
+        password: hashedPasswordValue
+    }
+  })
 
   req.session.user = newUser;
 
-  res.json(newUser);
-});
+  res.json(newUser)
+})
 
-router.put("/:id", async (req, res) => {
-  const { id } = req.params;
-  const { username, email, password } = req.body;
+router.put('/:id', async(req, res) => {
+  const { id } = req.params
+  const {  username, email, password  } = req.body
 
   const updatedUser = await prisma.user.update({
     where: { id: parseInt(id) },
     data: {
-      username,
-      email,
-      password,
-    },
-  });
+        username, 
+        email, 
+        password
+    }
+  })
 
-  res.json(updatedUser);
-});
+  res.json(updatedUser)
+})
 
-router.delete("/:id", async (req, res) => {
-  const { id } = req.params;
+router.delete('/:id', async(req, res) => {
+  const { id } = req.params
 
   const deletedUser = await prisma.user.delete({
-    where: { id: parseInt(id) },
-  });
-
-  res.json(deletedUser);
-});
-
-router.post("/login", async (req, res) => {
-  const { username, password } = req.body;
-
-  try {
-    // Find the user by username
-    const user = await prisma.user.findFirst({ where: { username: username } });
-
-    if (!user) {
-      return res.status(401).json({ error: "Invalid username or password" });
+    where: { id: parseInt(id) }
+  })
+
+  res.json(deletedUser)
+
+})
+
+router.post('/login', async (req, res) => {
+    const { username, password } = req.body;
+
+    try {
+      // Find the user by username
+      const user = await prisma.user.findFirst({ where: { username: username } });
+  
+      if (!user) {
+        return res.status(401).json({ error: 'Invalid username or password' });
```

Comment:
potentially: make the error message more clear. Here we can throw "user now found".

Reply:
**_To Do:_**
Re-work error messages